### DISCUSSION TITLE
Speed up sftp. How can far2l utilize rsync?

### DISCUSSION THREAD

**Discussion by singalen** [elfmz/far2l#926] [Ideas]
I was copying not too many (~1000) tiny (under 1K) files onto a slow sftp server, and it took ages (about 20 min).
Is it because sftp needs to open a connection and do protocol handshakes for each file?
Wouldn't `rsync` be much faster? I believe it sends all the files in-band in one connection and with a single handshake.
Or is there something else to fix up?

**Comment by elfmz** [comment]
Connection - no, but kind of handshake - yes. open file, read file, close file..
Try scp, there're rumors it can be faster:)
But of course best is to archive/tarball them and get as single file.
